sqlglot/tokens.py

```diff
@@ -833,6 +863,42 @@ def _scan_string(self, quote):
         self._add(TokenType.STRING, text)
         return True
 
+    def _scan_bit_string(self, bit_string):
+        bs_end = self._BIT_STRINGS.get(bit_string)
+        if bs_end is None:
+            return False
+
+        self._advance(len(bit_string))
+        text = self._extract_string(bs_end)
+        if bs_end != " ":
+            self._advance(len(bs_end) - 1)
+
+        try:
+            self._add(TokenType.BIT_STRING, f"{int(text, 2)}")
+        except ValueError:
+            raise RuntimeError(
+                f"Bit string contains invalid characters from {self._line}:{self._start}"
+            )
+        return True
+
+    def _scan_hex_string(self, hex_string):
```

Comment:
this looks the same as bit string, should we turn this into one method with args?

Reply:
Combined